### PR TITLE
[controller] Support ble connect using discriminator in python cli

### LINKED ISSUE #4063: CHIP BLE name in raspi cannot be set, it shows ubuntudesktop
CHIP BLE name in raspi cannot be set, it shows ubuntudesktop, instead of showing CHIP-1383 or CHIP-3840

### PR DESCRIPTION
**Pull request by erjiaqing**
#### Problem
We does not support ble connect using discriminator, which does match chip spec and is not convenient when device name is not CHIP-XXXX.

 #### Summary of Changes
Add discriminator support

#### Testing

```
ble-scan
ble-connect 1383
```
or
```
ble-scan-connect 1383
```

 Fixes #4063

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: src/controller/python/chip/ChipBluezMgr.py [hunk 1]
```diff
@@ -1060,20 +1075,37 @@ def disconnect_bg_implementation(self, **kwargs):
             self.service.destroy()
             self.service = None
 
+    def get_peripheral_devIdInfo(self, peripheral):
+        if not peripheral.ServiceData:
+            return None
+        servDataDict = dict(peripheral.ServiceData)
```
Comment (andy31415) [inline src/controller/python/chip/ChipBluezMgr.py:1081]: can you iterate over ServiceData directly without creating a separate dictionary only for the purpose of iteration?

Happens a few more times below. what type is ServiceData? is it a tuple array (in which case you can iterate normally) or something else?

File: src/controller/python/chip/ChipBluezMgr.py [hunk 2]
```diff
@@ -948,11 +950,20 @@ def dump_scan_result(self, device):
...
         self.logger.info("{0:<10}{1}".format("RSSI =", device.RSSI))
         self.logger.info("{0:<10}{1}".format("address =", device.Address))
-        self.logger.info(
-            "ADV data: " + ("".join([str(i) for i in dict(device.ServiceData).keys()]))
-            if device.ServiceData
-            else ""
-        )
+
+        if device.ServiceData:
+            serviceDataDict = dict(device.ServiceData)
```
Comment (andy31415) [inline src/controller/python/chip/ChipBluezMgr.py:955]: Why do you need to make ServiceData a dictionary?

File: src/controller/python/chip/ChipCoreBluetoothMgr.py [hunk 3]
```diff
@@ -220,8 +243,12 @@ def shouldLoop(self, should_tuple):
...
 
-            for peripheral in self.peripheral_list:
-                if should_tuple[3] and str(peripheral._.name) == should_tuple[3]:
+            for peripheral in self.peripheral_adv_list:
+                if should_tuple[3] and str(peripheral.peripheral._.name) == should_tuple[3]:
+                    result = False
+                    break
+                devIdInfo = peripheral.getPeripheralDevIdInfo()
+                if devIdInfo and should_tuple[3] and str(devIdInfo.discriminator) == should_tuple[3]:
                     result = False
```
Comment (andy31415) [inline src/controller/python/chip/ChipCoreBluetoothMgr.py:252]: `return False` (without single break) is much more pythonic. 

code shoulld likely look like:

```python

if time_expired:
  return False

status = should_tuple[0]

if status == "ready":
    return (not self.ready_condition)
elif status == "scan":
   # .... for loops for name checks
   return True
elif status == "connect":
   return (not self.loop_condition)
.....

return False
```

Comment (erjiaqing) [reply src/controller/python/chip/ChipCoreBluetoothMgr.py:252]: Aggree

File: src/controller/python/chip/ChipBluezMgr.py (hunk 1 above)
Comment (erjiaqing) [reply src/controller/python/chip/ChipBluezMgr.py:1081]: the service data is a dbus.Dictionary
I will go through the python dbus doc and then cleanup these code in this pr.

**Review by yunhanw-google — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by jelderton — APPROVED** [review]

**Pull request merged**